src/utils/request-api.js

```diff
@@ -160,7 +161,17 @@ function requestAPI (config, options, callback) {
   })
 
   if (options.files) {
-    stream.pipe(req)
+    if (options.progress && typeof options.progress === 'function') {
+      const progressStream = new Transform({
+        transform: (chunk, encoding, cb) => {
+          options.progress(chunk.byteLength)
+          cb(null, chunk)
+        }
+      })
+      stream.pipe(progressStream).pipe(req)
+    } else {
+      stream.pipe(req)
+    }
```

Comment:
@Kubuxu how does go-ipfs tell the CLI how much 'data' was ingested files? Isn't it returning a ndjson update?

Reply:
I am not sure but I am not sure what exactly  you are asking for.

> how much 'data' was ingested files?

Reply:
@Kubuxu when running a `ipfs add BIGFILE`, how does the CLI know how much data has been processed by go-ipfs?

Reply:
Following object is being set by the daemon to the CLI as file is being digested:
```
type AddedObject struct {
	Name  string
	Hash  string `json:",omitempty"`
	Bytes int64  `json:",omitempty"`
	Size  string `json:",omitempty"`
}
```
It causes update of the progress bar.

Reply:
Just confirmed with @whyrusleeping, go-ipfs http-api does indeed send a ndjson stream of updates when adding files if --progress is passed.

We can support streaming both in node and in the browser with stream-http (like we already do). 

@bmordan your PR needs to change to capture that and instead of measuring on bytes sent, measure on bytes reported back. Could you change to that, please?

Reply:
@diasdavid OK I'm having a look at this.